test/end2end.py

```diff
@@ -158,11 +166,33 @@ def test_hanging_file(self):
             "--setmpi impirun",
             "--output %s" % f_out,
             "--output-check-timeout 2",
+            "--disable-output-check-fatal",
             "hostname",
             ])
         ec, out = run_simple(cmd)
         self.assertEqual(ec, 0, "Command exited normally: exit code %s; output: %s" % (ec, out))
 
-        regex = re.compile("WARNING: mympirun has been running for .* seconds without seeing any output.")
+        regex = re.compile("mympirun has been running for .* seconds without seeing any output.")
         self.assertTrue(len(regex.findall(out)) == 1, "Pattern '%s' found in: %s" % (regex.pattern, out))
 
+
+    def test_hanging_fatal(self):
+        """Test fatal hanging check when program has no output"""
+        no_output_mpirun = '\n'.join([
+            "#!/bin/bash",
+            "sleep 4",
+        ])
+
+        install_fake_mpirun('mpirun', self.tmpdir, txt=no_output_mpirun)
+        cmd = ' '.join([
```

Comment:
add a comment here and mention that `--output-check-fatal` is enabled by default